pkg/tagger/collectors/kubernetes_metadata_mapper.go

```diff
@@ -103,6 +99,26 @@ func (c *KubeMetadataCollector) getTagInfos(pods []*kubelet.Pod) []*TagInfo {
 	return tagInfo
 }
 
+func (c *KubeMetadataCollector) getMetadaNames(metadataByNsPods apiv1.NamespacesPodsStringsSet, po *kubelet.Pod) ([]string, error) {
+	if !c.clusterAgentEnabled {
+		metadataNames, err := apiserver.GetPodMetadataNames(po.Spec.NodeName, po.Metadata.Namespace, po.Metadata.Name)
+		if err != nil {
+			err = fmt.Errorf("Could not fetch cluster level tags of pod: %s, %v", po.Metadata.Name, err)
+		}
+		return metadataNames, err
+	}
+
+	if c.dcaClient.ClusterAgentVersion.Major >= 1 && c.dcaClient.ClusterAgentVersion.Minor >= 3 {
+		return metadataByNsPods[po.Metadata.Namespace][po.Metadata.Name].List(), nil
```

Comment:
maybe we can fetch `metadataByNsPods` in this block as well to avoid checking the conditions again in the main func ?

Reply:
what we can do it to check if the map is empty. if empty it means we didn't retrieved the data before

Reply:
yes we need to do some checks, `GetPodsMetadataForNode` can be `nil` if there's an `err` in `GetPodsMetadataForNode`

Reply:
PR updated with additional nil checks